lib/northbound.c

```diff
@@ -549,7 +569,31 @@ int nb_candidate_edit(struct nb_config *candidate,
 		dnode = lyd_new_path(candidate->dnode, ly_native_ctx,
 				     xpath_edit, (void *)data->value, 0,
 				     LYD_PATH_OPT_UPDATE);
-		if (!dnode && ly_errno) {
+		if (dnode) {
+			/*
+			 * create dependency
+			 *
+			 * dnode returned by the lyd_new_path may be from a
+			 * different schema, so we need to update the nb_node
+			 */
+			nb_node = dnode->schema->priv;
+			if (nb_node->dep_cbs.get_dependency_xpath) {
+				nb_node->dep_cbs.get_dependency_xpath(
+					dnode, dep_xpath);
+
+				ly_errno = 0;
+				dep_dnode = lyd_new_path(candidate->dnode,
```

Comment:
Consider one scenario 
```
c t
ip route 1.1.1.1/32 ens33
```

In this configuration you will create a default dnode in frr-vrf yang table.

Now I have deleted the route
```
c t
no ip route 1.1.1.1/32 ens33
```
Now default dnode is not getting deleted from frr-vrf yang table.

Reply:
This issue is not directly related to this PR. This already happens for any other configuration that needs default VRF dnode. For example, when you configure vni in zebra.
I'm not sure that this is an issue at all. Nothing wrong with always having a configuration node for the default VRF as this VRF always exists.
Anyway, this should not be a blocker for this fix and can be discussed and fixed separately.

Reply:
ok, we can discuss and fixed separately.